src/site/shared/state/Header/reducers.ts

```diff
@@ -13,7 +13,8 @@ export function headerReducer(state = initialState, action: AllSharedActions): H
         case OPEN_HEADER_MENU_ID:
             return {
                 ...state,
-                curMenu: action.menu
+                // Close the menu by clicking on its icon while it is open
+                curMenu: ((state.curMenu === action.menu) ? ("none") : (action.menu))
```

Comment:
This totally works, but in general, the reducers and their state should be as "dumb" as possible. They pretty much directly take in actions and cause a logical change in state. And so when I, as a programmer, call "OpenHeaderMenu", I expect the menu to be open and not toggle. I think the better change would be in `src/site/shared/containers/Header/Right/Dropdown.tsx` and change it so that when you click it and it's already open, it closes it instead.

Reply:
The onclick behavior is defined for each separate dropdown rather than unified in ```Dropdown.tsx```. Should each separate dropdown be modified so that it calls a new function such as ```ToggleHeaderMenu```, or should just the name of the dropdown be passed into ```Dropdown.tsx``` so that Dropdown can decide whether to call OpenHeaderMenu or CloseHeaderMenus? (or is there some other way that I'm missing?)